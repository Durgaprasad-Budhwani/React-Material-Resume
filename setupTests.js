/**
 * Created by Durgaprasad Budhwani on 12/14/2017.
 */
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Element.prototype.getBoundingClientRect = function(){
    return {
        y: -100,
        top: -100
    }
};

configure({ adapter: new Adapter() });