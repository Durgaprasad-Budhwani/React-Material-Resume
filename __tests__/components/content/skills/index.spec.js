/**
 * Created by durgaprasad on 10/10/17.
 */
/* global expect,it,describe,afterEach */
import React from 'react';
import SkillsContainer, {Skills} from '../../../../src/components/content/skills';
import renderer from 'react-test-renderer';
import fetchMock from 'fetch-mock';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import '../../../../setupTests';

describe('<Skills />', () => {
    const classes = {
        root: {},
        button: {},
        container: {}
    };
    
    const skill = {name: 'abc', completed: 10};
    
    afterEach(() => {
        fetchMock.restore();
    });
    
    it('renders <SkillsContainer />', () => {
        let skills = [ skill ];
        fetchMock.mock('skills.json', skills);
        
        const tree = renderer.create(
            <SkillsContainer
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('renders <Skills />', () => {
        let skills = [ skill ];
        fetchMock.mock('skills.json', skills);
        const wrapper = mount(
            <Skills
                classes={classes}
            />
        );
        let instance = wrapper.instance();
        let skillWrapper = shallow(instance._renderSkill(skill, 1));
        expect(toJson(skillWrapper)).toMatchSnapshot();
    });
});
