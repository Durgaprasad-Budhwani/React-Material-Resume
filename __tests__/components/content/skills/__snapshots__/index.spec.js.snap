// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders skill bar 1`] = `
<div
  align="stretch"
  className="MuiGrid-typeContainer-4 MuiGrid-direction-xs-column-6 MuiGrid-justify-xs-center-20"
  id="skills"
>
  <span
    className="SectionTitle-sectionTitle-93"
  >
    Professional Skills
  </span>
  <div
    className="MuiPaper-root-94 MuiPaper-shadow2-98 MuiPaper-rounded-95 Skilss-root-1"
  >
    <div
      className="MuiGrid-typeContainer-4 MuiGrid-typeItem-5 MuiGrid-spacing-xs-16-25 MuiGrid-justify-xs-space-between-22 MuiGrid-grid-xs-12-40"
    />
  </div>
</div>
`;
